crates/editor/src/editor.rs

```diff
@@ -7687,25 +7689,63 @@ impl Editor {
         self.select_larger_syntax_node_stack = stack;
     }
 
-    fn runnable_display_rows(
-        &self,
-        range: Range<Anchor>,
+    fn refresh_runnables(&mut self, cx: &mut ViewContext<Self>) -> Task<()> {
+        let display_snapshot = self.display_map.update(cx, |map, cx| map.snapshot(cx));
+        let project = self.project.clone();
+        cx.spawn(|this, mut cx| async move {
+            let Some(project) = project else {
+                return;
+            };
+            if project
+                .update(&mut cx, |this, _| this.is_remote())
+                .unwrap_or(true)
+            {
+                // Do not display any test indicators in remote projects.
+                return;
+            }
+            let new_rows =
+                cx.background_executor()
+                    .spawn({
+                        let snapshot = display_snapshot.clone();
+                        async move {
+                            Self::fetch_runnable_ranges(&snapshot, Anchor::min()..Anchor::max())
```

Comment:
Does it make a difference to query only for the viewport vs. all?

Reply:
I think so, but OTOH I don't think we have access to viewport there?